cpp/src/strings/regex/regcomp.cpp

```diff
@@ -372,41 +415,77 @@ class regex_parser {
           break;
         }
         case 'w': {
-          if (_id_cclass_w < 0) { _id_cclass_w = _prog.add_class(cclass_w); }
-          _cclass_id = _id_cclass_w;
+          if (is_ascii(_flags)) {
```

Comment:
All the cases seem very repetitive and it would be nice if you could use a templated solution to write them just once. There also appear to be some very slight differences in some cases though (e.g. adding the newline characters to the character class) so maybe that's not feasible. Just planting the idea in case you think it could help.

Reply:
Thanks. I suspect there will be more changes in this area and I don't want to refactor this too soon and then have to undo the refactoring later.